app/src/main/java/com/andrew/tasky/presentation/agenda/AgendaFragment.kt

```diff
@@ -1,64 +1,279 @@
+import kotlinx.coroutines.launch
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
 
 class AgendaFragment : Fragment(R.layout.fragment_agenda) {
 
     private lateinit var navController: NavController
     private lateinit var viewModel: AgendaViewModel
-    private lateinit var binding: FragmentAgendaBinding
+    private lateinit var fragmentAgendaBinding: FragmentAgendaBinding
+    private lateinit var currentDateAndTime: LocalDateTime
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         navController = Navigation.findNavController(view)
-        viewModel = ViewModelProvider(this).get(AgendaViewModel::class.java)
-        binding = FragmentAgendaBinding.bind(view)
-
-        var agendaItemList = mutableListOf(
-            AgendaItem(AgendaItemType.EVENT, false, "Meeting",
-                "Project meeting with Abby", "3:00", "Mar 5"),
-            AgendaItem(AgendaItemType.REMINDER, false, "Take out trash",
-                "Take trash from inside to outside", "4:00", "Mar 5"),
-            AgendaItem(AgendaItemType.TASK, true, "Code",
-                "Finish coding lesson", "5:00", "Mar 5"),
-            AgendaItem(AgendaItemType.EVENT, false, "Run",
-                "Go for an evening run", "6:00", "Mar 5"),
+        viewModel = ViewModelProvider(this)[AgendaViewModel::class.java]
+        fragmentAgendaBinding = FragmentAgendaBinding.bind(view)
+        currentDateAndTime = LocalDateTime.now()
+
+        val agendaItemList = mutableListOf(
+            AgendaItem(
+                AgendaItemType.EVENT, false, "Meeting",
+                "Project meeting with Abby", LocalDateTime.of(2022, 10, 3, 8, 22),
+                LocalDateTime.of(2022, 10, 3, 8, 32)
+            ),
+            AgendaItem(
+                AgendaItemType.REMINDER, false, "Take out trash",
+                "Take trash from inside to outside", LocalDateTime.of(2022, 10, 3, 8, 37)
+            ),
+            AgendaItem(
+                AgendaItemType.TASK, true, "Code",
+                "Finish coding lesson", LocalDateTime.of(2022, 10, 3, 9, 22)
+            ),
+            AgendaItem(
+                AgendaItemType.EVENT, false, "Run",
+                "Go for an evening run", LocalDateTime.of(2022, 10, 3, 12, 22),
+                LocalDateTime.of(2022, 10, 15, 14, 22)
+            ),
         )
 
         val adapter = AgendaItemAdapter(agendaItemList)
 
-        binding.agendaItemRecyclerView.adapter = adapter
-        binding.agendaItemRecyclerView.layoutManager = LinearLayoutManager(view.context)
+        fragmentAgendaBinding.agendaItemRecyclerView.adapter = adapter
+        fragmentAgendaBinding.agendaItemRecyclerView.layoutManager =
+            LinearLayoutManager(view.context)
+
+        subscribeToObservables()
+        setOnClickListeners()
+    }
+
+    private fun <T> Fragment.collectLatestLifecycleFlow(
```

Comment:
I'd put this in a separate file and make it public since you'll likely need this in all of your Fragments. Here you'll never expect such functionality